spacy/pipeline/spancat.py

```diff
@@ -183,27 +302,52 @@ def __init__(
         vocab: Vocab,
         model: Model[Tuple[List[Doc], Ragged], Floats2d],
         suggester: Suggester,
+        # XXX Not sure what's the best default name when it can both be spancat
+        # and spancat_singlelabel
         name: str = "spancat",
         *,
+        single_label: bool = False,
```

Comment:
I don't think you need a separate setting for `single_label`, it could just be `max_positive = 1`, right?

Reply:
ok your right!

Reply:
I just thought its nice to make it like very explicit, because it does something different when its `single_label`, but I'll change it.

Reply:
Oh actually! I didn't do that because you might want to use a `Logistic` classifier with a `threshold` even when `max_positive=1` right? Or for simplicity we should not allow this and `max_positive=1` unambiguously means `Softmax` classifier?

Reply:
If you use the `spancat` factory with `max_positive=1` then we have a `Logistic` classifier and no negative-label. If `single_label=True` we run the `Softmax` classifier with the negative label. So currently its not the same model. If `max_positive=1` should imply `Softmax` classifier, then when you use the `spancat` factory and say `max_positive=1` we should actually exit with an error?